packages/lib/server/queries/teams/index.ts

```diff
@@ -124,43 +133,39 @@ export async function getTeamWithMembers(args: {
   });
 
   if (!team) return null;
-  const members = await Promise.all(
-    team.members.map(async (obj) => {
-      const calendarCredentials = getCalendarCredentials(obj.user.credentials);
-
-      const { connectedCalendars } = await getConnectedCalendars(
-        calendarCredentials,
-        obj.user.selectedCalendars,
-        obj.user.destinationCalendar?.externalId
-      );
-      const connectedApps = obj.user.credentials
-        .map(({ app, id }) => {
-          const appMetaData = getAppFromSlug(app?.slug);
-
-          if (app?.categories.includes(AppCategories.calendar)) {
-            const externalId = connectedCalendars.find((cal) => cal.credentialId == id)?.primary?.email;
-            return { name: appMetaData?.name, logo: appMetaData?.logo, slug: appMetaData?.slug, externalId };
-          }
-          return { name: appMetaData?.name, logo: appMetaData?.logo, slug: appMetaData?.slug };
-        })
-        .sort((a, b) => (a.slug ?? "").localeCompare(b.slug ?? ""));
-      // Prevent credentials from leaking to frontend
-      const {
-        credentials: _credentials,
-        destinationCalendar: _destinationCalendar,
-        selectedCalendars: _selectedCalendars,
-        ...rest
-      } = {
-        ...obj.user,
-        role: obj.role,
-        accepted: obj.accepted,
-        disableImpersonation: obj.disableImpersonation,
-        avatar: `${WEBAPP_URL}/${obj.user.username}/avatar.png`,
-        connectedApps,
-      };
-      return rest;
-    })
-  );
+
+  // This should improve performance saving already app data found.
+  const appDataMap = new Map();
+
+  const members = team.members.map((obj) => {
+    return {
+      ...obj.user,
+      role: obj.role,
+      accepted: obj.accepted,
+      disableImpersonation: obj.disableImpersonation,
+      avatar: `${WEBAPP_URL}/${obj.user.username}/avatar.png`,
+      connectedApps: obj.user.credentials.map((cred) => {
+        const appSlug = cred.app?.slug;
+        let appData = appDataMap.get(appSlug);
+
+        if (!appData) {
+          appData = getAppFromSlug(appSlug);
+          appDataMap.set(appSlug, appData);
+        }
+
+        const isCalendar = cred?.app?.categories.includes("calendar");
+        const externalId = isCalendar ? cred.destinationCalendars[0]?.externalId : undefined;
```

Comment:
Getting [0] index should give us the main destinationCalendar, but I don't think this would be more than 1 item here since it's 1 calendar per credential.